src/util/prolog/prologSession.ts

```diff
@@ -53,38 +72,53 @@ function getDependencyGraph(
 		return graph;
 }
 
+/** Executes the prolog programs and queries, fetches the raw results and creates the PrologResult */
 export class PrologSession {
 
-		public readonly session: any;
-
-		constructor(program: string, defaultPath: string = __dirname, fileName?: string, limit?: number) {
-				program = PSM.removeNonFunctionalities(program);
-				program = resolveImports(program, defaultPath, fileName);
-
-				this.session = new type.Session(limit);
-				this.session.consult((PSM.removeModuleDeclarations(program)));
-		}
-
-		public async executeQuery(query: string): Promise<PrologResult> {
-				this.session.query(query);
-
-				const rawResults: string[] = [];
-				let result: string;
-
-				do {
-						result = await this.getNextAnswer();
-						rawResults.push(result);
-				} while (result[result.length - 1] !== ".");
-
-				return new PrologResult(rawResults);
-		}
-
-		private getNextAnswer(): Promise<string> {
-				return new Promise<string>((resolve) => {
-						this.session.answers((x: any) => {
-								resolve(format_answer(x));
-						});
-				});
-		}
+	public readonly session: any;
+
+	/**
+		* @param program - The program code to run
+		* @param defaultPath - Only needed for forwarding methode internally
+		* @param fileName - Only needed for forwarding methode internally
+		* @param maxSteps - The maximum amount of resolving steps to do before aborting a query
+		*/
+	constructor(program: string, defaultPath: string = __dirname, fileName?: string, maxSteps?: number) {
+			program = PSM.removeNonFunctionalities(program);
+			program = resolveImports(program, defaultPath, fileName);
+
+			this.session = new type.Session(maxSteps);
+			this.session.consult((PSM.removeModuleDeclarations(program)));
+	}
+
+	/**
+		* Executes the query on the program and creates the result
+		* @param query - The query to run on the given codebase
+		* @return The result for the query
+		*/
+	public async executeQuery(query: string): Promise<PrologResult> {
+			this.session.query(query);
+
+			const rawResults: string[] = [];
+			let result: string;
+
+			do {
+					result = await this.getNextAnswer();
+					rawResults.push(result);
+			} while (result[result.length - 1] !== ".");
+
+			return new PrologResult(rawResults);
+	}
+	/**
+		* Receives the answers for the query set by set
+		* @return The next result set
+		*/
+	private getNextAnswer(): Promise<string> {
+			return new Promise<string>((resolve) => {
+					this.session.answers((x: any) => {
+							resolve(format_answer(x));
```

Comment:
Nit: "x" is somewhat meaningless, but I think it's not really necessary to rename it, since the context describes it well enough already. But we should refrain from using such variablenames more often.

Reply:
Agreed